PlasmaAnchor/anchor.md

```diff
@@ -0,0 +1,113 @@
+    // Msg contains all subjectively finalized data 
+    // since last objectively finalized data
+    RootchainBlockHash [][]byte
+    Deposits           []Deposit
+    Exits              []Exit
+}
+```
+
+For validator A with subjectively finalized block `bA` and subjectively finalized deposit `nA`.
+
+Then ValidatorA would send `AnchorMsg` with RootchainBlockHash including BlockHashes `B+1` through `bA`, Deposits `N+1` through `nA`, and all new subjectively finalized exits that are not objectively finalized.
+
+If the Validator does not agree with the Anchor Chain
+i.e., there exists some blocknumber k < N s.t. Anchor's kth block has a different blockhash then the validator rootchain node's kth blockhash; then the validator does not send an `AnchorMsg` until his rootchain view converges with the AnchorChain.
+
+### Handling
+
+##### Ante
+Antehandler makes sure msg sender is a sidechain validator and validator has not already sent `AnchorMsg` in this block
+
+##### Handler
+
+The AnchorMsg Handler then stores all the above data into a transient store called `Subjective`.
+
+##### EndBlocker
+
+At the end of the block, `endBlocker` loads all the lists validators inputted which is stored in subjective
+
+`endBlocker` then starts with the first element of each list and checks if more than 2/3 of validators agree on a single value. If they do, it is added to the appropriate objectively finalized store. Then endBlocker continues through each element of the list objectively finalizing it by adding the data to the checkpointed stores until there isn't a 2/3 quorum on the next value.
+
+Thus, `endBlocker` calculates the Maximum Continuous Intersection of BlockHashes and Deposits independently and adds them to their respective stores.
+
+NOTE: Exits have no notion of being continuous so it is just the Intersection of Exits among a 2/3 quorum of validators.
+
+Since there are a limited number of sidechain validators and the lists are limited to being at most the length of new data that has been added to rootchain since last objectively finalized checkpoint, this isn't an infeasibly expensive calculation to happen in `endBlocker`.
+
+### Subjective Finality
+
+In order to preserve the notion of subjective finality, validators define a parameter in their anchor node's `config.toml` file called `subjective-finality`. This is how many blocks the validator believes need to be added after a piece of data before that data is presumed finalized by the validator. The Anchor node will then poll the rootchain node continuously and pull all relevant data that has been subjectively finalized according to the above parameter but not objectively finalized by Anchor and construct an `AnchorMsg` submitting that data to Anchor chain.
+
+To tune their `subjective-finality` parameter, validators simply stop their anchor node, update their `config.toml` with their preferred `subjective-finality` bound, and restart the anchor node.
+
+### Catastrophic Failure
```

Comment:
Future work should look to address this, would be irrelevant for Plasma but worth doing in general case